### PR TITLE
Add rust libraries page

### PR DESCRIPTION
**Pull request by svrnm**
Introduces a page about instrumentation libraries for rust. It's still only some basic content, but it's a start:-)

### CODE REVIEW

**Comment by cartermp** [comment]
Several formatting/spelling failures here

File: content/en/docs/languages/rust/libraries.md
Comment (hdost) [inline content/en/docs/languages/rust/libraries.md:20]: It might be good to add https://github.com/tokio-rs/tracing-opentelemetry << Since in the rust community you have many crates which are already instrumented with the `tracing` crate from `tokio-rs` this allows for wiring those into OpenTelemetry.

Comment (svrnm) [reply content/en/docs/languages/rust/libraries.md:20]: Good call out, thank you, I can add a few words around that too!

**Comment by svrnm** [comment]
> Several formatting/spelling failures here

I developed this PR alongside #3830 and decided then to split them, unfortunately the adding of "Arctix" to textlintrc seems to lead to issues, so this PR is blocked by the other one

File: content/en/docs/languages/rust/libraries.md
Comment (svrnm) [inline content/en/docs/languages/rust/libraries.md:22]: @hdost like this? wdyt?

File: content/en/docs/languages/rust/libraries.md
Comment (chalin) [inline content/en/docs/languages/rust/libraries.md:15]:
```suggestion
For example, the
[instrumentation library for Actix Web](https://crates.io/crates/actix-web-opentelemetry)
```

File: content/en/docs/languages/rust/libraries.md
Comment (chalin) [inline content/en/docs/languages/rust/libraries.md:17]:
```suggestion
[metrics](/docs/concepts/signals/metrics/) based on the inbound HTTP requests.
```

File: content/en/docs/languages/rust/libraries.md
Comment (chalin) [inline content/en/docs/languages/rust/libraries.md:25]:
```suggestion
For a list of available instrumentation libraries, see the
[registry](/ecosystem/registry/?language=js&component=instrumentation).
```

File: content/en/docs/languages/rust/libraries.md
Comment (chalin) [inline content/en/docs/languages/rust/libraries.md:20]:
```suggestion
If your app or one of your dependencies makes use of the
[tracing](https://crates.io/crates/tracing) crate, use
```

File: content/en/docs/languages/rust/libraries.md
Comment (chalin) [inline content/en/docs/languages/rust/libraries.md:21]:
```suggestion
[tracing-opentelemetry](https://crates.io/crates/tracing-opentelemetry) to
```

File: content/en/docs/languages/rust/libraries.md
Comment (chalin) [reply content/en/docs/languages/rust/libraries.md:22]: Change "adding" to "add".

File: content/en/docs/languages/rust/libraries.md
Comment (cijothomas) [inline content/en/docs/languages/rust/libraries.md:25]:
```suggestion
[registry](/ecosystem/registry/?language=rust&component=instrumentation).
```

File: content/en/docs/languages/rust/libraries.md
Comment (cijothomas) [reply content/en/docs/languages/rust/libraries.md:21]: tracing-opentelemetry is already listed in the registry as an instrumentation - don't think we need to specially call it out. (Agree it is very popular, but there are some interop issues, not fully sorted out..)

Comment (svrnm) [reply content/en/docs/languages/rust/libraries.md:21]: @cijothomas I added this based on a comment from @hdost: https://github.com/open-telemetry/opentelemetry.io/pull/3828#discussion_r1460956728

Since you are the rust experts, I would leave it to you to have it or not

File: content/en/docs/languages/rust/libraries.md
Comment (svrnm) [reply content/en/docs/languages/rust/libraries.md:22]:
```suggestion
If your app or one of your dependencies is making use of the create
[`tracing`](https://crates.io/crates/tracing) you can use
[`tracing-opentelemetry`](https://crates.io/crates/tracing-opentelemetry) for
add OpenTelemetry interoperability.
```

Comment (svrnm) [inline content/en/docs/languages/rust/libraries.md:22]:
```suggestion
add OpenTelemetry interoperability.
```

File: content/en/docs/languages/rust/libraries.md
Comment (cijothomas) [inline content/en/docs/languages/rust/libraries.md:5]:
```suggestion
description: How to use instrumentation libraries
```

File: content/en/docs/languages/rust/libraries.md
Comment (cijothomas) [reply content/en/docs/languages/rust/libraries.md:21]: No need to explicitly list, given it is already listed in the registry. It is not something maintained by the OpenTelemetry organization. (If this changes in the future, we can add it).

File: content/en/docs/languages/rust/libraries.md
Comment (cijothomas) [reply content/en/docs/languages/rust/libraries.md:5]: maybe stick with what we already have in other pages 
https://github.com/open-telemetry/opentelemetry.io/blob/main/content/en/docs/languages/net/libraries.md?plain=1#L2-L4

Comment (svrnm) [reply content/en/docs/languages/rust/libraries.md:5]: I picked that description because people might not know what an instrumentation library is, bay having the title "Using instrumentation libraries" this would feel redundant

Comment (cijothomas) [reply content/en/docs/languages/rust/libraries.md:5]: nvm, its inconsistent across languages...https://github.com/open-telemetry/opentelemetry.io/blob/main/content/en/docs/languages/js/libraries.md?plain=1#L5

File: content/en/docs/languages/rust/libraries.md
Comment (svrnm) [reply content/en/docs/languages/rust/libraries.md:21]: let me remove it then once again

File: content/en/docs/languages/rust/libraries.md
Comment (svrnm) [reply content/en/docs/languages/rust/libraries.md:5]: Eventually we should have descriptions for all the pages, but this is yet another consistency challenge.

If you have strong objections against this description, I can remove it, otherwise we can keep it and refine it later

File: content/en/docs/languages/rust/libraries.md
Comment (cartermp) [reply content/en/docs/languages/rust/libraries.md:21]: If memory serves me, there's a ton of adoption of the tokio tracing crate and several folks in the Rust community recommend using that with the OTel exporter. Not a "now" thing but we should probably try to quantify that a bit, because if the community leans towards something we should also lean into it.

Comment (cijothomas) [reply content/en/docs/languages/rust/libraries.md:21]: Yes, it is an extremely popular crate. (we in OTel Rust, does some special handling for supporting that crate in our sdk :))

All I said was there is no need to explicitly recommend/call-out that, especially since there are unresolved issues with the combination of using tracing + otel. The otel rust maintainers and tokio-tracing maintainers are working to resolve them, and once the resolution is there, we can do an explicit call out. (it is unknown what that resolution would be... it could be that tracing-opentelemetry is merged into otel itself or vice-versa or something else. Whatever the resolution will be, it'll be first called out in OTel rust repo itself, and then we can move it to opentelemetry.io/docs too.)

Comment (cartermp) [reply content/en/docs/languages/rust/libraries.md:21]: Awesome, thanks @cijothomas

Comment (cartermp) [reply content/en/docs/languages/rust/libraries.md:21]: Let's go with @cijothomas' suggestion for now

**Review by cijothomas — APPROVED** [review]

**Review by cartermp — APPROVED** [review]

**Pull request merged**
